scripts/skills/special/rf_naked.nut

```diff
@@ -0,0 +1,38 @@
+this.rf_naked <- ::inherit("scripts/skills/skill", {
+	m = {
+		BraveryMult = 0.5,
+		ExcludedBackgrounds = [
+			"background.wildman",
+			"background.barbarian"
+		]
+	},
+	function create()
+	{
+		this.m.ID = "special.rf_naked";
+		this.m.Name = "Naked";
+		this.m.Description = "This character\'s private parts are fully exposed, making it a rather nervous situation to be in.";
+		this.m.Icon = "skills/rf_naked.png";
+		this.m.Type = ::Const.SkillType.Special | ::Const.SkillType.StatusEffect;
+		this.m.IsSerialized = false;
+		this.m.IsHidden = true;
+	}
+
+	function isHidden()
+	{
+		return !this.isEnabled();
+	}
+
+	function onUpdate( _properties )
+	{
+		if (this.isEnabled())
+		{
+			_properties.BraveryMult *= this.m.BraveryMult;
+		}
+	}
+
+	function isEnabled()
+	{
+		local actor = this.getContainer().getActor();
+		return ::MSU.isKindOf(actor, "player") && ::MSU.isNull(actor.getBodyItem()) && this.m.ExcludedBackgrounds.find(actor.getBackground().getID()) == null)
```

Comment:
The `::MSU.isKindOf(actor, "player")` should be partially removed to allow this effect to work on anyone, if we or a mod decide to add it to other characters.
Right now it is even redundant because we only add this effect to player characters anyways.

I suggest do write the check like this:
```
	function isEnabled()
	{
		local actor = this.getContainer().getActor();
		return ::MSU.isNull(actor.getBodyItem()) && (!::MSU.isKindOf(actor, "player") || this.m.ExcludedBackgrounds.find(actor.getBackground().getID()) == null);
	}
```

Reply:
Alright I'll make the change. The condition would have to be a bit more involved in that case though, because `getBackground()` only exists in `player`. So I'd have to put that behind an `isKindOf` check.

Reply:
If we trust the interpreter to evaluate the first argument of || first, always, then we can trust that actor is a player, the moment that we check for `this.m.ExcludedBackgrounds.find(actor.getBackground().getID()) == null`

Anyways, your rework of this function is good

Reply:
Ah I didn't see that you had it in an `||` there. Yeah that would've worked.